server/fleet/app.go

```diff
@@ -707,6 +711,27 @@ func (f *Features) ApplyDefaults() {
 	f.EnableHostUsers = true
 }
 
+func (f *Features) Clone() (interface{}, error) {
+	return f.Copy(), nil
+}
+
+func (f *Features) Copy() *Features {
+	if f == nil {
+		return nil
+	}
+
+	var clone Features
+	clone = *f
```

Comment:
Shouldn't it manually copy the `DetailQueryOverrides` map?

Reply:
not necessary per my understanding! see https://go.dev/play/p/MazLl2_fZmj

Reply:
oh, maybe you're thinking of copying each pointer in the map? like:

```go
mapCopy := new(map[string]*string, len(f.DetailQueryOverrides))

for k, v :=  range f.DetailQueryOverrides {
    mapCopy[k] = &strings.Clone(*v)
}
```

Reply:
Yeah (same address here on the playground sample):
```go
fmt.Printf("address of copy map: %p\n", t.Y["foo"])
fmt.Printf("address of copy map: %p\n", t2.Y["foo"])
```

Reply:
great catch! thanks! what do you think? [7b59d65](https://github.com/fleetdm/fleet/pull/14291/commits/7b59d658371522ecae8a314d19233b040fd2f193)